### PR TITLE
Set up CI (build docker image)

### PR DESCRIPTION
**Pull request by dmgav**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+language: python
+sudo: false
+
+services:
+  - docker
+
+env:
+  global:
+    - GH_REF: github.com/NSLS-II/scipy-binder.git
```
Comment (danielballan) [inline .travis.yml:9]: What is this used for?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    - python: 3.7
+      env: BUILD_DOCS=true SUBMIT_CODECOV=true NUMPY=1.17
+
+python:
+  - 3.7
+
+env:
+  - NUMPY=1.17
+
+before_install:
```
Comment (danielballan) [inline .travis.yml:28]: I think I mentioned as I was thinking this through that we could install conda this way. But, as I continued thinking it through, I realized that we really want to test that `repo2docker  .` works end to end, and that will take care of running conda for us (inside a docker container). Thus I think this is unnecessary.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+env:
+  global:
+    - GH_REF: github.com/NSLS-II/scipy-binder.git
+
+cache:
+  directories:
+    - $HOME/.cache/pip
+    - $HOME/.ccache
+
+matrix:
```
Comment (danielballan) [inline .travis.yml:16]: What's intent here?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  fast_finish: true
+  include:
+    - python: 3.7
+      env: BUILD_DOCS=true SUBMIT_CODECOV=true NUMPY=1.17
+
+python:
+  - 3.7
+
+env:
```
Comment (danielballan) [inline .travis.yml:25]: And here?

**Comment by danielballan** [comment]
As I mentioned in chat, this trick might be useful:

https://github.com/bluesky/tutorials/blob/658a50bd1af3fed6c9b7344f329dcd88803b90b1/.travis.yml#L28

**Comment by dmgav** [comment]
I planned to make the script work first and then remove the unnecessary code. I will try the `.keep_alive.sh` trick.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  global:
+
+install:
+  - export GIT_FULL_HASH=`git rev-parse HEAD`
+  - bash .keep_alive.sh &  # Generate output periodically.
+  - pip install jupyter-repo2docker
+  - pip list
+
+script:
+  - repo2docker . --no-run
```
Comment (danielballan) [inline .travis.yml:17]: Oops, I led you astray here. I think it has to be

```
repo2docker --no-run .
```

Comment (dmgav) [reply .travis.yml:17]: @danielballan  Let's see if it helps

**Comment by danielballan** [comment]
Failed with:

```
The job exceeded the maximum time limit for jobs, and has been terminated.
```

We might be able to shave off a little time by being clever, but overall I think the build time will _grow_ significantly over the next couple months as we add new Python packages and JupyterLab extensions, so I think this is a losing battle.

Maybe it's worth investigating GitHub Actions and/or Azure Pipelines, something that @stuartcampbell has suggested in any case for automating deployment of the updated images.

**Comment by danielballan** [comment]
Superseded by https://github.com/NSLS-II/scipy-binder/pull/15 which uses the GH Actions solution we have used in tutorials and other repos.

**Pull request closed without merging**